### PR TITLE
Reduce bundle size

### PR DESCRIPTION
**Pull request by jorisre**
👋🏻 

Many things in this pull request:

1. Update dependencies
2. Improve jest config
3. Since [V10](https://github.com/okonet/lint-staged/releases/tag/v10.0.0) of lint-staged, we don't have to add `git add` anymore
4. Add [microbundle](https://github.com/developit/microbundle) _(replace custom rollup config)_
5. Add [size-limit](https://github.com/ai/size-limit) + github action
6. Remove old rollup configuration
7. Remove IE11 config _(not necessary anymore)_
8. Remove development bundle  _(not necessary anymore)_
9. Multiples optimisations

Tested on IE11 ✅ 

Before: **649 B**
After: **264 B** (~59% saved)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -17,10 +18,24 @@
...
+    {
+      "path": "dist/index.umd.js",
+      "limit": "270 B"
+    }
+  ],
   "scripts": {
     "clean": "rimraf dist",
     "prebuild": "npm run clean",
-    "build": "node rollup/writeCjsEntryFile.js && rollup -c",
+    "build": "microbundle build --jsx React.createElement",
```
Comment (bluebill1049) [inline package.json:38]: That's cool!

File: src/ErrorMessage.tsx [hunk 2]
```diff
@@ -25,25 +25,18 @@ const ErrorMessage = <
   }
 
   const { message: messageFromRegister, types } = error;
-  const props = {
-    ...rest,
+  const props = Object.assign({}, rest, {
     children: messageFromRegister || message,
-  };
+  });
```
Comment (bluebill1049) [inline src/ErrorMessage.tsx:30]: Is this actually less code after bundle? We should do that in hook form as well, every byte counts! ❤️

**Comment by bluebill1049** [comment]
wow, this `microbundle` is magic.

File: src/ErrorMessage.tsx (hunk 2 above)
Comment (jorisre) [reply src/ErrorMessage.tsx:30]: yes it saves some bytes :)

Comment (bluebill1049) [reply src/ErrorMessage.tsx:30]: That's awesome!

**Comment by bluebill1049** [comment]
@jorisre you can branch off from master now :) you are a team member. (in case you didn't know)

**Comment by jorisre** [comment]
Size limit fails because of https://github.com/andresz1/size-limit-action/issues/33

**Review by bluebill1049 — APPROVED** [review]
LGTM!

**Comment by bluebill1049** [comment]
Are you happy to merge this in? we can release a beta version of it.

**Comment by jorisre** [comment]
> Are you happy to merge this in? we can release a beta version of it.

Yeah ! you can merge it :)

**Comment by bluebill1049** [comment]
Functional test: https://codesandbox.io/s/react-hook-form-v6-errormessage-dn4y8 Pass 🟢

**Comment by bluebill1049** [comment]
The amazing result: https://bundlephobia.com/result?p=@hookform/error-message@0.0.5-beta.1

Screen Shot 2020-11-22 at 8 38 26 pm

Thank you very much @jorisre

**Comment by bluebill1049** [comment]
I will tweet this once the current register API poll is over on Twitter! Exciting and publishing the new patch now.

**Comment by jorisre** [comment]
It's strange on bundlephobia bundle size is 483B whereas in size-limit [264 B](https://github.com/react-hook-form/error-message/blob/master/package.json#L24)

**Comment by bluebill1049** [comment]
> It's strange on bundlephobia bundle size is 483B whereas in size-limit [264 B](https://github.com/react-hook-form/error-message/blob/master/package.json#L24)

I am already super impressed with 483B, LOL!

**Comment by kotarella1110** [comment]
@jorisre @bluebill1049 awesome PR 😎

**Pull request merged**